simulator-ui/src/main/webapp/app/entities/scenario-execution/scenario-execution.model.ts

```diff
@@ -1,45 +1,15 @@
 import dayjs from 'dayjs/esm';
 
 import { IMessage } from 'app/entities/message/message.model';
+import { ITestResult } from 'app/entities/test-result/test-result.model';
 
 export interface IScenarioExecution {
   executionId: number;
   startDate?: dayjs.Dayjs | null;
   endDate?: dayjs.Dayjs | null;
   scenarioName?: string | null;
-  status?: 'UNKNOWN' | 'RUNNING' | 'SUCCESS' | 'FAILED' | null;
-  errorMessage?: string | null;
+  testResult?: ITestResult | null;
   scenarioMessages?: IMessage[] | null;
 }
 
 export type NewScenarioExecution = Omit<IScenarioExecution, 'executionId'> & { executionId: null };
```

Comment:
Why is this called **New**ScenarioExecution?

Also there are some more types in _scenario-execution.service.ts_ and the type hierarchy gets a bit complex. Why are all those types needed?

Reply:
I think that was a leftover from earlier days. I've removed it in d1e164d318f2c158a79b3059fdc8b021c2b5666b, as it was unused except in tests.